source/extensions/filters/http/file_system_buffer/fragment.h

```diff
@@ -102,6 +102,11 @@ class Fragment {
 private:
   size_t size_;
   absl::variant<MemoryFragment, StorageFragment, ReadingFragment, WritingFragment> data_;
+
+  // Disabling copy and move constructors for safety because long-lived fragment pointers
+  // are used.
+  Fragment(const Fragment&) = delete;
+  Fragment(Fragment&&) = delete;
```

Comment:
You an inherit from NonCopyable (search code) if you want. Feel free to do this in a follow up if you like.